### PR TITLE
[FIX] web: add vertical align to buttons in list view

### PR DESCRIPTION
**Pull request by clesgow**
Description of the issue/feature this PR addresses:
Following the changes in 902fcba, it added a new vertical alignment for elements in a list view.
The issue is that since `o_list_button` are excluded from the new rules, when we have a line thicker than one line, the buttons won't be aligned with the other fields anymore.
Same issue with `o_list_record_remove` and `o_handle_cell` that won't be aligned with the other fields either.

Example in the Purchase app
Current behavior before PR:
image

Desired behavior after PR is merged:
image

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by pparidans** [comment]
@Whenrow  Looks like @mano-odoo worked on a very similar topic in https://github.com/odoo/odoo/pull/138915 and https://github.com/odoo/enterprise/pull/49823 .

The approach not being exactly the same, it's probably interesting to discuss it before merging either solution.

Invoke @stefanorigano for advice

**Comment by stefanorigano** [comment]
> Invoke @stefanorigano for advice

To align listView elements to the middle it's a long term goal, IMHO we can't achieve it right now because of `<table>` tag intrinsic limitation + inconsistencies on how we handle data (sometimes by truncating text, sometimes by wrapping elements).
image
One day we'll probably switch do CSS grid (and we'll face other type of issues 😅 ), but for now I'd suggest to use vertical align top. Using middle has proven to be problematic especially for selection checkboxes and drag icons... every time we tried to address an issue with middle, there it was yet another failing scenario. 

In conclusion I'd go for mano's approach. It should fix this bug + others across the webclient.

**Comment by pparidans** [comment]
Fixed in https://github.com/odoo/odoo/pull/145320
